dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -191,16 +192,19 @@ void Trace::configureSink(const std::vector<TraceAppender>& appenders) {
 
     sink->set_formatter(fmt);
 
-    if (appenders[i].getTag() == "") {
+    if (appenderTag == "") {
       sink->set_filter(severity >= appenders[i].getLvlFilter() && !expr::has_attr(tag_attr) && thread_attr == currentId);
     } else {
-      sink->set_filter(severity >= appenders[i].getLvlFilter() && tag_attr == appenders[i].getTag() && thread_attr == currentId);
+      sink->set_filter(severity >= appenders[i].getLvlFilter() && tag_attr == appenderTag && thread_attr == currentId);
     }
     logging::core::get()->add_sink(sink);
-    if (appenders[i].isPersistant())
-      traceSink.persistantSinks.push_back(sink);
-    else
-      traceSink.sinks.push_back(sink);
+    if (appenders[i].isPersistant()) {
+      assert(traceSink.persistantSinks.find(appenderTag) == traceSink.persistantSinks.end());
+      traceSink.persistantSinks.insert({appenderTag, sink});
```

Comment:
Are e sure this tag is always unique? What happens if we add the same tag but with different level filters in 2 different files? Moreover we need to make sure dyn-algo and dfl logs are still working after this modification